src/bci_build/registry.py

```diff
@@ -49,20 +49,32 @@ def registry_prefix(*, is_application: bool) -> str:
     @staticmethod
     def build_version(base_version, container: "BaseContainerImage") -> str:
         """Return the build version to set for this container build."""
-        container_version: str = container.tag_version
-        # if container_version is a numeric version and not a macro, then
-        # version.parse() returns a `Version` object => then we concatenate
-        # it with the existing build_version
-        # for non PEP440 versions, we'll get an exception and just return
-        # the parent's classes build_version
-        try:
-            packaging.version.parse(str(container_version))
-            stability_suffix: str = ""
-            if container._stability_suffix:
-                stability_suffix = "." + container._stability_suffix
-            return f"{base_version}.{container_version}{stability_suffix}"
-        except packaging.version.InvalidVersion:
-            return base_version
+
+        def _build_version_from_container_version(
+            container_version: int | str | None,
+        ) -> str | None:
+            # if container_version is a numeric version and not a macro, then
+            # version.parse() returns a `Version` object => then we concatenate
+            # it with the existing build_version
+            # for non PEP440 versions, we'll get an exception and just return
+            # None
+            try:
+                packaging.version.parse(str(container_version))
+                stability_suffix: str = ""
+                if container._stability_suffix:
+                    stability_suffix = "." + container._stability_suffix
+                return f"{base_version}.{container_version}{stability_suffix}"
+            except packaging.version.InvalidVersion:
+                return None
+
+        # first try to use the containers full version, if that fails, pick
+        # tag_version and if that also fails, fall back to the parent's classes
+        # version
```

Comment:
so that means switching from a `%%..`%%` based version to another version will change the behavior. is that really a good idea? seems very dangerous to me. 

I think the current setup of "version" being the %%placeholder and "tag_version" being a placeholder-free "stable" version is good enough and more predictable in behavior. I know we fscked it up for kiwi but that has been fixed and there is now a check that tag_version needs to be set if there is a placeholder in version. we could enforce it even more to have tag_version always set which I think would make the behavior more predicatble.